### PR TITLE
#39 :arrow_up: update default python version from 3.9.16 to 3.11.4 to…

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**
… ensure latest binary installer is available; :sparkles: catch poetry check error and perform remediation

### CODE REVIEW

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InitializeHabushuMojo.java [hunk 1]
```diff
@@ -21,27 +21,35 @@ public class InitializeHabushuMojo extends AbstractHabushuMojo {
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InitializeHabushuMojo.java:23]: I (Informational): recommend hiding whitespace

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InitializeHabushuMojo.java [hunk 2]
```diff
@@ -21,27 +21,35 @@ public class InitializeHabushuMojo extends AbstractHabushuMojo {
...
-			"Poetry package version set to %s in pyproject.toml does not align with expected POM-derived version of %s",
-			currentPythonPackageVersion, expectedPythonPackageVersion));
-	    }
-
-	}
+        getLog().info("Validating Poetry-based project structure...");
+        PoetryCommandHelper poetryHelper = createPoetryCommandHelper();
+        try {
+            poetryHelper.execute(Arrays.asList("check"));
+        } catch (HabushuException e) {
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InitializeHabushuMojo.java:28]: Seems to occur when you update the minimum python version in your `pyproject.toml`.  It is easy to remiediate, so we catch the error and have poetry resync `pypoject.toml` and `poetry.lock`.  If it were to fail during this, the error would be throw to the user.

File: .github/workflows/maven.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ jobs:
         steps:
             - uses: actions/checkout@v3
             - name: Setup Pyenv
```
Comment (d-ryan-ashcraft) [inline .github/workflows/maven.yaml:24]: I (Informational): Need newer version of pyenv to install python 3.11.4

**Review by carter-cundiff — APPROVED** [review]

**Review by Ross-Jeffrey — COMMENTED** [review]
Change seems fine in my limited experience. I've reviewed the build check and based on that clearing I'd say this is an approved change. I do still see a few spots in the habushu repo that reference python version of 3.10.4 like 
cleanHabushuMojo.java and ValidatePyenvAndPoetryMojo.java, unless I'm misunderstanding things?

**Comment by d-ryan-ashcraft** [comment]
> Change seems fine in my limited experience. I've reviewed the build check and based on that clearing I'd say this is an approved change. I do still see a few spots in the habushu repo that reference python version of 3.10.4 like cleanHabushuMojo.java and ValidatePyenvAndPoetryMojo.java, unless I'm misunderstanding things?

Those are examples of how to specify a non-default version of Python - I _think_ it is useful they are different than the default version, but could be convinced otherwise as well.

**Pull request merged**